motor/providers/ssh/session.go

```diff
@@ -201,14 +204,48 @@ func prepareConnection(pCfg *providers.Config) ([]ssh.AuthMethod, []io.Closer, e
 		}
 	}
 
-	if len(sshSigners) > 0 {
-		auths = append(auths, ssh.PublicKeys(sshSigners...))
-	}
-
 	// if no credential was provided, fallback to ssh-agent and ssh-config
 	if len(pCfg.Credentials) == 0 {
 		sshSigners = append(sshSigners, signers.GetSignersFromSSHAgent()...)
 	}
 
+	// If no keys were loaded so far, then attempt loading the default keys in the home directory
+	if len(sshSigners) == 0 {
```

Comment:
This is adding more implicit behavior. We should add a new credential type for ssh config and only load it when the credential type is set. The cli behaviour should stay implicit and therefore we need to add ssh_config and ssh_agent as default there.